### PR TITLE
Introduce funding section to provide project grant information

### PR DESCRIPTION
**Pull request by sven1103**
A new custom field for funding information is introduced to provide grant information, which enriches the available project information to store.

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/ProjectFormLayout.java [hunk 1]
```diff
@@ -74,6 +77,23 @@ public ProjectFormLayout() {
         .bind((ProjectInformation::getProjectObjective),
             ProjectInformation::setProjectObjective);
 
+    fundingField = new FundingField("Funding Information");
+    binder.forField(fundingField)
+        .withValidator(field -> {
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/ProjectFormLayout.java:82]: The element you are passing into the lambda is not the field but rather its value. So I suggest renaming it to `fundingInformation` or `funding`

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/ProjectFormLayout.java [hunk 2]
```diff
@@ -74,6 +77,23 @@ public ProjectFormLayout() {
...
+              if (field == null) {
+                return true;
+              }
+              if (!field.getLabel().isBlank()) {
+                return !field.getReferenceId().isBlank();
+              } else if (!field.getReferenceId().isBlank()) {
+                return !field.getLabel().isBlank();
+              }
+              return true;
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/ProjectFormLayout.java:91]: Should be split up into different validators to allow for different error messages depending on the cause. Right now it is not possible to say something like `please provide the funding reference id` when only funding label is filled out.

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/Funding.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * For example:
+ * <ul>
+ *   <li>SFB (label)</li>
+ *   <li>SFB 1101 (reference id)</li>
+ * </ul>
+ *
+ * @since 1.0.0
+ */
+@Embeddable
+public class Funding {
```
Comment (KochTobi) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/Funding.java:19]: Please provide matching `equals` and `hashCode` methods.

File: vaadinfrontend/frontend/themes/datamanager/components/dialog.css [hunk 4]
```diff
@@ -182,6 +182,12 @@ vaadin-dialog-overlay::part(title) {
   align-items: baseline;
 }
 
+.create-project-dialog .input-fields {
+   display: flex;
+   gap: 1em;
+   align-items: baseline;
+ }
+
```
Comment (KochTobi) [inline vaadinfrontend/frontend/themes/datamanager/components/dialog.css:190]: why is this necessary?

**Review by KochTobi — CHANGES_REQUESTED** [review]

**Review by KochTobi — CHANGES_REQUESTED** [review]
Hi @sven1103 
The code looks good. However, I would argue that the validator can be split into two giving us more flexibility without increasing complexity much. 
I further have a question to the intended functionality. When funding information is provided at any point, it is impossible to remove it again. Instead an empty funding entry is stored. Is this desired?

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/general/funding/FundingField.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    super();
+    setLabel(fieldLabel);
+    this.label = new TextField("Grant", "e.g. SFB");
+    this.referenceId = new TextField("Grant ID", "e.g. SFB 1101");
+    this.binder = new Binder<>();
+    layoutComponent();
+    configureBinder();
+  }
+
+  private void configureBinder() {
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/general/funding/FundingField.java:34]: can be removed

File: vaadinfrontend/frontend/themes/datamanager/components/dialog.css (hunk 4 above)
Comment (KochTobi) [reply vaadinfrontend/frontend/themes/datamanager/components/dialog.css:190]: Isn't that done already?

Comment (KochTobi) [reply vaadinfrontend/frontend/themes/datamanager/components/dialog.css:190]:
```
.create-project-dialog .code-and-title,
.contact-field .input-fields {
  display: flex;
  gap: 1em;
  align-items: baseline;
}
```

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/general/funding/FundingField.java [hunk 6]
```diff
@@ -45,6 +34,10 @@ private static Div layoutFundingInput(TextField label, TextField referenceId) {
     return layout;
   }
 
+  private void layoutComponent() {
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/general/funding/FundingField.java:37]: You could inline this method.
Something like `add(layoutInputs)` is readable enough and when looking for the layout can be detected directly.

**Review by KochTobi — APPROVED** [review]
beautiful LGTM
